packages/orchestration/src/orchestration-response.ts

```diff
@@ -51,6 +55,56 @@ export class OrchestrationResponse {
     return choice?.message?.content;
   }
 
+  /**
+   * Messages that can be used for subsequent prompts as message history.
+   * @param choiceIndex - The index of the choice to parse.
+   * @returns A list of all messages.
+   */
+  getAllMessages(choiceIndex = 0): ChatMessages {
+    const messages: ChatMessage[] = (
+      this.data.module_results.templating ?? []
+    ).map(message =>
+      this.isMultiChatMessage(message)
+        ? this.handleMultiChatMessage(message as MultiChatMessage)
+        : this.handleSingleChatMessage(message as SingleChatMessage)
+    );
+
+    const content = this.getChoices().find(c => c.index === choiceIndex)
```

Comment:
[q] How do we handle cases when the assistant response has empty content and `tool_calls`? It should retain that structure.